### PR TITLE
prevent timeout if debugger is attached

### LINKED ISSUE #14871: Don't timeout SignalR clients if a debugger is attached
HubConnectionContext.Start/CheckClientTimeout currently carry on as usual even if a debugger is attached.

### PR DESCRIPTION
**Pull request by Kahbazi**
Fixes #14871

### CODE REVIEW

File: src/SignalR/server/Core/src/HubConnectionContext.cs [hunk 1]
```diff
@@ -554,17 +554,20 @@ internal void StartClientTimeout()
 
         private void CheckClientTimeout()
         {
-            // If it's been too long since we've heard from the client, then close this
-            if (DateTime.UtcNow.Ticks - Volatile.Read(ref _lastReceivedTimeStamp) > _clientTimeoutInterval)
+            if (!Debugger.IsAttached)
```
Comment (davidfowl) [inline src/SignalR/server/Core/src/HubConnectionContext.cs]: I'm a huge fan of reducing nesting by exiting early. So in this case:

```C#
if (Debugger.IsAttached)
{
    return;
}
```

Comment (Kahbazi) [reply src/SignalR/server/Core/src/HubConnectionContext.cs]: Done!

**Review by analogrelay — COMMENTED** [review]
Seems reasonable to me, we've done similar stuff on the server.

**Review by davidfowl — APPROVED** [review]

**Review by BrennanConroy — APPROVED** [review]

**Comment by BrennanConroy** [comment]
Thanks @Kahbazi !

**Pull request merged**